### PR TITLE
Assessment and question JSON validation

### LINKED ISSUE #684: add check that endDate cannot be before startDate
When validating allowAccess stanzas, check that the EndDate is after the startDate.

### PR DESCRIPTION
**Pull request by trombonekenny**
Adds JS validation of assessment and question files based on `lib/course-db:checkInfoValid`

Adds allowAccess startDate and endDate checking (closes #684)
Fixes return logic bugs in loops
Adds tests for these conditions

### CODE REVIEW

File: lib/course-db.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ var path = require('path');
 var _ = require('lodash');
 var fs = require('fs');
 var async = require('async');
+var moment = require('moment-timezone');
```
Comment (mwest1066) [inline lib/course-db.js]: Why not just use plain `moment` here? I don't think we need the timezone capabilities?

File: tests/testLoadCourse.js [hunk 2]
```diff
@@ -30,3 +30,187 @@ describe('courseDB.loadFullCourse()', function() {
...
+
+            fs.writeFile(questionFilename, JSON.stringify(questionJson), function(err) {
+                if (ERR(err, callback)) return;
+                callback(null);
+            });
+        });
+    });
+
+    after('removing test files', function(callback) {
+        //console.log('Removing test files');
```
Comment (mwest1066) [inline tests/testLoadCourse.js]: remove debugging

File: tests/testLoadCourse.js [hunk 3]
```diff
@@ -30,3 +30,187 @@ describe('courseDB.loadFullCourse()', function() {
...
 });
+
+describe('courseDB.loadFullCourse() on brokenCourse', function() {
+
+    var courseDir = 'tests/testLoadCourse/brokenCourse';
+    var assessmentFilename = `${courseDir}/courseInstances/Fa18/assessments/quiz1/infoAssessment.json`;
+    var questionFilename = `${courseDir}/questions/basicV3/info.json`;
+
+    beforeEach('write correct infoAssessment and question', function(callback) {
+        //console.log('Resetting brokenCourse files');
```
Comment (mwest1066) [inline tests/testLoadCourse.js]: remove debugging

File: lib/course-db.js (hunk 1 above)
Comment (trombonekenny) [reply lib/course-db.js]: *shrug* Habit.

**Pull request merged**